TradeItIosTicketSDK2/TradeItAccountManagementViewController.swift

```diff
@@ -76,22 +76,28 @@ class TradeItAccountManagementViewController: TradeItViewController, TradeItAcco
             withMessage: "Are you sure you want to unlink your account and remove all the associated data?",
             withActionTitle: "Unlink",
             onAlertActionTapped: { () -> Void in
-                TradeItSDK.linkedBrokerManager.unlinkBroker(self.linkedBroker)
-
-                // If the last linked broker was just unlinked then we need to use the TradeItBrokerManagementViewController
-                // that preceeds this view controller in the nav stack to launch the broker linking flow so that we can pop
-                // this view controller off the nav stack without the TradeItLinkBrokerUIFlow being garbage collected
-                if let navController = self.navigationController,
-                    let parentBrokerManagementView = navController.viewControllers[safe: navController.viewControllers.count - 2] as? TradeItBrokerManagementViewController {
-                    if TradeItSDK.linkedBrokerManager.linkedBrokers.count == 0 {
-                        parentBrokerManagementView.linkBrokerUIFlow.presentLinkBrokerFlow(
-                            fromViewController: navController,
-                            showWelcomeScreen: true,
-                            oAuthCallbackUrl: TradeItSDK.oAuthCallbackUrl
-                        )
+                TradeItSDK.linkedBrokerManager.unlinkBroker(
+                    self.linkedBroker,
+                    onSuccess: {
+                        // If the last linked broker was just unlinked then we need to use the TradeItBrokerManagementViewController
+                        // that preceeds this view controller in the nav stack to launch the broker linking flow so that we can pop
+                        // this view controller off the nav stack without the TradeItLinkBrokerUIFlow being garbage collected
+                        if let navController = self.navigationController,
+                            let parentBrokerManagementView = navController.viewControllers[safe: navController.viewControllers.count - 2] as? TradeItBrokerManagementViewController {
+                            if TradeItSDK.linkedBrokerManager.linkedBrokers.count == 0 {
+                                parentBrokerManagementView.linkBrokerUIFlow.presentLinkBrokerFlow(
+                                    fromViewController: navController,
+                                    showWelcomeScreen: true,
+                                    oAuthCallbackUrl: TradeItSDK.oAuthCallbackUrl
+                                )
+                            }
+                            navController.popViewController(animated: true)
+                        }
+                    },
+                    onFailure: { error in
+                        self.alertManager.showError(error, onViewController: self)
```

Comment:
If for some reason we hit the case you mention below where `let error = TradeItErrorResult(title: "UnlinkBroker error")` then it's possible that we aren't handling the condition in the comments above either. I think in this case, if the broker doesn't exist locally anymore it should just be `onSuccess`